src/js/contentScript.js

```diff
@@ -3,26 +3,58 @@
  *  JavaScript file to be injected into specified pages
  *******************************************************************************/
 
+ /**
+  * Gets all elements with #INSPECTED_TAGS tags and returns filtered results
+  * @return {Array} Array of DOM elements to inspect
+  * @protected
+  **/
+ const _getDomElements = function ()
+ {
+     let divElements = $(INSPECTED_TAGS).toArray();
+     divElements = divElements.filter(function (elem) {
+         return elem.innerText.trim();
+     });
+     if (divElements.length > MAX_ELEMENTS)
+     {
+         divElements.length = MAX_ELEMENTS;
+     }
+     return divElements;
+ };
+
+
+ /**
+  * Shows the element in the page
+  * @param $elem The jQuey element to hide
+  **/
+ const _showDomELement = function ($elem)
+ {
+     var oldText = $elem.attr(INITIAL_DATA_ATTR);
+     $elem.removeAttr(INITIAL_DATA_ATTR);
+     $elem.html(oldText);
+     $elem.removeClass(OFFENSIVE_WARNING);
+ };
+
+
+ /**
+  * Iterates over the web page's elements and show them
+  * @return {int} The number of offensive words found in the page
+  **/
+ const _revertElements = function ()
+ {
+     let divElements = _getDomElements();
```

Comment:
Instead of iterating through all the dom elements again, you can just use [jQuery class selector](https://api.jquery.com/class-selector/)
to fetch elements with the class `offensiveWarning`
And put _getDomElements back in the inspector after that